docs/l2/withdrawals.md

```diff
@@ -1,24 +1,123 @@
-  - `to`: the address in L1 that is allowed to claim the funds (this is decided by the user as part of a withdraw transaction). This comes from the regular `to` field on the Withdraw transaction (i.e. we are reusing that field with a slightly different meaning; what it means here is "the address that can claim the funds on L1").
-  - `amount`: the amount of money withdrawn (i.e. the `msg.value` of the transaction).
-  - `tx_hash`: the transaction hash in the L2 block it was included in. This will be important for claiming the withdrawal as it will require a merkle proof to be provided along with the index on the tree.
-- As part of the L1 `commit` transaction, the sequencer will send the list of all `WithdrawLog`s on the EIP 4844 blob (i.e. as a section of the state diffs) and the `WithdrawLogsRoot` as calldata as part of the public input to the proof. The contract will then:
-  - Verify that the withdraw logs passed on the blob are the correct ones (this is done as part of the proof of equivalence protocol explained below).
-  - Store the `WithdrawLogsRoot` on a mapping `(batchNumber -> LogsRoot)`
-- For users to complete their withdraw process and receive funds on the L1, they need to call a `claimWithdraw(l2WithdrawalTxHash, claimedAmount, withdrawalBatchNumber, withdrawalLogIndex)` function on the common bridge, where `merkleProof` is an inclusion proof of the withdraw log to the root of the merkle tree the contract has stored. The contract will then do the following:
-  - Check that the `batchNumber` corresponds to a committed and verified batch.
-  - Check that this withdrawal has not been already claimed.
-  - Retrieve the `withdrawLogsRoot` from the given `batchNumber`.
-  - Verify the merkle proof given by the user, passing the proof, the root, and the `tx_hash`.
-  - If any check above failed, revert. If all checks passed, send the appropriate funds to the user, then set the `withdrawLog` as claimed.
-  - After the withdrawal is sent, we mark it as claimed so it cannot be claimed twice.
+This document contains a detailed explanation of how asset withdrawals work.
+
+## Native ETH withdrawals
+
+This section explains step by step how native ETH withdrawals work.
+
+On L2:
+
+1. The user sends a transaction calling `withdraw(address _receiverOnL1)` on the `CommonBridgeL2` contract, along with the amount of ETH to be withdrawn.
+2. The bridge sends the withdrawn amount to the burn address.
+3. The bridge calls `sendMessageToL1(bytes32 data)` on the `L1Messenger` contract, with `data` being:
+
+    ```solidity
+    bytes32 data = keccak256(abi.encodePacked(ETH_ADDRESS, ETH_ADDRESS, _receiverOnL1, msg.value))
+    ```
+
+4. `L1Messenger` emits an `L1Message` event, with the address of the L2 bridge contract and `data` as topics.
```

Comment:
I think we are missing an "Off chain" explanation of what happens with this event.